src/structure/matrix.rs

```diff
@@ -1772,6 +2378,20 @@ impl Add<Matrix> for Matrix {
                     }
                 }
                 result
+
+                // Note: Parallel version - how to add the implementation below?
+                // let mut result = par_matrix(self.data.clone(), self.row, self.col, self.shape);
+                // result
+                //     .data
+                //     .par_iter_mut()
+                //     .enumerate()
+                //     .for_each(|(idx, value)| {
+                //         let i = idx / self.col;
+                //         let j = idx % self.col;
+
+                //         *value += other[(i, j)];
+                //     });
+                // result
```

Comment:
The commented part above is the parallel version. The above commented part is the parallelized part. How can we have `impl Add<Matrix> for Matrix` and have both seq and par versions?
If it is not possible, then can we have the parallel version by default??

This applies to `Neg ` and `Sub ` too.